packages/wallets/trezor/src/adapter.ts

```diff
@@ -0,0 +1,169 @@
+
+    get connecting() {
+        return this._connecting;
+    }
+
+    get readyState() {
+        return this._readyState;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            if (this._readyState !== WalletReadyState.Loadable) throw new WalletNotReadyError();
+
+            this._connecting = true;
+
+            const connectUrl = this._connectUrl && this._connectUrl + (this._connectUrl.endsWith('/') ? '' : '/');
+
+            await TrezorConnect.init({
+                manifest: {
+                    email: 'maintainers@solana.foundation',
+                    appUrl: 'https://github.com/solana-labs/wallet-adapter',
+                },
+                lazyLoad: true,
+                ...(this._connectUrl
+                    ? {
+                          connectSrc: connectUrl,
+                          iframeSrc: connectUrl,
+                      }
+                    : {}),
+            });
+
+            let result;
+            try {
+                result = await TrezorConnect.solanaGetPublicKey({
+                    path: this._derivationPath,
+                });
+            } catch (error: any) {
+                throw new WalletPublicKeyError(error?.message, error);
+            }
+
+            if (!result.success) {
+                throw new WalletPublicKeyError(result.payload?.error, result.payload);
+            }
+
+            const publicKey = result.payload.publicKey;
+
+            TrezorConnect.on(DEVICE_EVENT, (event: any) => {
+                if (event.type === DEVICE.DISCONNECT) {
+                    this._disconnected();
+                }
+            });
+
+            this._publicKey = new PublicKey(Buffer.from(publicKey, 'hex'));
+
+            this.emit('connect', this._publicKey);
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        } finally {
+            this._connecting = false;
+        }
+    }
+
+    async disconnect(): Promise<void> {
+        this._publicKey = null;
+
+        try {
+            await TrezorConnect.dispose();
+        } catch (error: any) {
+            this.emit('error', new WalletDisconnectionError(error?.message, error));
+        }
+
+        this.emit('disconnect');
+    }
+
+    async signTransaction<T extends Transaction | VersionedTransaction>(transaction: T): Promise<T> {
+        try {
+            const publicKey = this._publicKey;
+            if (!publicKey) throw new WalletNotConnectedError();
+
+            const serializedTransaction = isVersionedTransaction(transaction)
+                ? transaction.message.serialize()
+                : transaction.serializeMessage();
+
+            let result;
+            try {
+                result = await TrezorConnect.solanaSignTransaction({
+                    path: this._derivationPath,
+                    serializedTx: Buffer.from(serializedTransaction).toString('hex'),
```

Comment:
Same here, we never use hex for data on Solana. Base64 strings are okay if it cannot be a Uint8Array.